auth/internal/transport/s2a_test.go

```diff
@@ -93,3 +94,37 @@ func TestMTLSConfigExpiry(t *testing.T) {
 	// Let the MTLS config expire before running other tests.
 	time.Sleep(1 * time.Second)
 }
+
+func TestIsGoogleS2AEnabled(t *testing.T) {
+	testCases := []struct {
+		name      string
+		useS2AEnv string
+		want      bool
+	}{
+		{
+			name:      "true",
+			useS2AEnv: "true",
+			want:      true,
+		},
+		{
+			name:      "false",
+			useS2AEnv: "false",
+			want:      false,
+		},
+		{
+			name:      "empty",
+			useS2AEnv: "",
+			want:      false,
+		},
+	}
+
+	for _, tc := range testCases {
+		if tc.useS2AEnv != "" {
+			os.Setenv(googleAPIUseS2AEnv, tc.useS2AEnv)
```

Comment:
You don't want to use os.Setenv is tests as it changes the state for the whole test run. Instead you can use: https://pkg.go.dev/testing#T.Setenv which restores the value after.

Reply:
Yes, I meant to use `t.Setenv`, thanks for catching this.